src/lib/input/input-wrapper.ts

```diff
@@ -0,0 +1,283 @@
+import {
+  Component,
+  Input,
+  Directive,
+  AfterContentInit,
+  ContentChild,
+  ContentChildren,
+  ElementRef,
+  QueryList,
+  ViewEncapsulation,
+  Optional,
+  Output,
+  EventEmitter
+} from '@angular/core';
+import {MdError, coerceBooleanProperty} from '../core';
+import {NgModel} from '@angular/forms';
+
+
+// Invalid input type. Using one of these will throw an MdInputWrapperUnsupportedTypeError.
+const MD_INPUT_INVALID_TYPES = [
+  'button',
+  'checkbox',
+  'color',
+  'file',
+  'hidden',
+  'image',
+  'radio',
+  'range',
+  'reset',
+  'submit'
+];
+
+
+const MD_INPUT_NEVER_EMPTY_TYPES = (() => {
+  let featureTestInput = document.createElement('input');
+  return [
+    'date',
+    'datetime',
+    'datetime-local',
+    'month',
+    'time',
+    'week'
+  ].filter(value => {
+    featureTestInput.setAttribute('type', value);
+    return featureTestInput.type === value;
+  });
+})();
```

Comment:
Move this to a separate `getSupportedInputTypes` function? Maybe under `core/platform`